### PR TITLE
remove panic from set_program_and_pages_data in common

### LINKED ISSUE #883: Remove panic from set_program
### File Location(s)

[common/src/lib.rs](https://github.com/gear-tech/gear/pull/882/files#diff-d38884f7391ae50f795707dbb281656e8afe38ea3ecac15a6ee80d52724017c0)

### Proposal

remove panic and make result

### PR DESCRIPTION
**Pull request by grishasobol**
Resolves #883 .

### CODE REVIEW

File: pallets/gear-program/Cargo.toml [hunk 1]
```diff
@@ -16,6 +16,7 @@ targets = ["x86_64-unknown-linux-gnu"]
 codec = { package = "parity-scale-codec", version = "3.1.2", default-features = false, features = ["derive"] }
 scale-info = { version = "2.0.1", default-features = false, features = ["derive"] }
 primitive-types = { version = "0.11.1", default-features = false, features = ["scale-info"] }
+log = "0.4"
```
Comment (breathx) [inline pallets/gear-program/Cargo.toml:19]: make std feature optional, please

**Review by breathx — APPROVED** [review]
👍

File: pallets/gear-program/src/pause.rs [hunk 2]
```diff
@@ -110,7 +110,12 @@ impl<T: Config> pallet::Pallet<T> {
 
         PausedPrograms::<T>::remove(program_id);
 
-        common::set_program_and_pages_data(program_id, paused_program.program, memory_pages);
+        if let Err(err) =
+            common::set_program_and_pages_data(program_id, paused_program.program, memory_pages)
+        {
+            log::error!("Wrong page with data number: {:?}", err.0);
```
Comment (ark0f) [inline pallets/gear-program/src/pause.rs:116]: It's better to implement `fmt::Display` for error

File: pallets/gear/src/manager.rs [hunk 3]
```diff
@@ -120,7 +120,7 @@ where
             state: common::ProgramState::Uninitialized { message_id },
         };
 
-        common::set_program_and_pages_data(program_id.into_origin(), program, Default::default());
+        common::set_program(program_id.into_origin(), program);
```
Comment (ark0f) [inline pallets/gear/src/manager.rs:123]: Are we ok with fact that function can panic?

Comment (grishasobol) [reply pallets/gear/src/manager.rs:123]: I think no. But it's not the problem of current PR.

File: pallets/gear-program/Cargo.toml (hunk 1 above)
Comment (grishasobol) [reply pallets/gear-program/Cargo.toml]: Ok

File: pallets/gear-program/src/pause.rs (hunk 2 above)
Comment (grishasobol) [reply pallets/gear-program/src/pause.rs]: ok

**Pull request merged**